src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorage.cs

```diff
@@ -134,7 +116,21 @@ static Subscriber Deserialize(string serializedForm)
             return new Subscriber(parts[0], endpointName);
         }
 
-        void Remove(string subscriber, MessageType messageType)
+        void StoreSubscription(Subscriber subscriber, MessageType messageType)
+        {
+            var toSend = new Message
+            {
+                Recoverable = true,
+                Label = Serialize(subscriber),
```

Comment:
Not introduced in this pull but the fact that we use the label, [max(250) chars](https://msdn.microsoft.com/en-us/library/ms703250(v=vs.85).aspx) , and that we now add the logical endpoint name to it we run a higher risk of hitting the size limit. I guess we should consider moving all the info to the body going forwards (separate discussion)